sql/wsrep_thd.cc

```diff
@@ -348,8 +348,16 @@ bool wsrep_bf_abort(const THD* bf_thd, THD* victim_thd)
 
   if (WSREP(victim_thd) && !victim_thd->wsrep_trx().active())
   {
-    WSREP_DEBUG("wsrep_bf_abort, BF abort for non active transaction");
-    wsrep_start_transaction(victim_thd, victim_thd->wsrep_next_trx_id());
+     WSREP_DEBUG("wsrep_bf_abort, BF abort for non active transaction");
+     switch (victim_thd->wsrep_trx().state()) {
+     case wsrep::transaction::s_aborting: /* fall through */
+     case wsrep::transaction::s_aborted:
+       WSREP_DEBUG("victim is aborting or has aborted");
+      return false;
+     default: break;
+     }
+
+     wsrep_start_transaction(victim_thd, victim_thd->wsrep_next_trx_id());
```

Comment:
I still don't understand why this is needed. A new wsrep transaction should always be started before THD accesses MDL or storage engine, so if a thd becomes a victim, it should always either:
* Have active transaction here, or
* Have been rolled back before BF thd enters here

Starting a wsrep transaction on behalf of another thread sounds wrong.

Reply:
I don't remember why this was needed, wsrep_start_transaction call was planted by the original 8.0 merging code drop in commit: f0d1f7db6f598aa0408ee01928278dba539565af
I will put debug assert before this call, and we can see if it fires in our testing